src/PDPVerifier.sol

```diff
@@ -493,21 +494,21 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         removeRoots(setId, removalsToProcess);
         // Bring added roots into proving set
         challengeRange[setId] = proofSetLeafCount[setId];
-        if (challengeEpoch - block.number < challengeFinality) {
+        if (challengeEpoch < block.number + challengeFinality) {
             revert("challenge epoch must be at least challengeFinality epochs in the future");
         }
         nextChallengeEpoch[setId] = challengeEpoch;
 
         // Clear next challenge epoch if the set is now empty.
-        // It will be re-set when new data is added.
+        // It will be re-set after new data is added and nextProvingPeriod is called.
         if (proofSetLeafCount[setId] == 0) {
             proofSetLastProvenEpoch[setId] = NO_PROVEN_EPOCH;
-            nextChallengeEpoch[setId] = 0;
+            nextChallengeEpoch[setId] = NO_CHALLENGE_SCHEDULED;
```

Comment:
I think we should revert if the prover passes in a non-zero `challengeEpoch` for a proofset that becomes empty after processing all their removals. It's not a valid path for the prover to take.

Reply:
Yeah I think this is a more consistent style

Reply:
But thinking more about it I prefer it the way it is.  By the time we get to the code to figure out if we should revert we've already done most of the work.  And the code has enough info to do the right thing in this case so it should do the right thing.  One thing that would be helpful though is being very explicit about this to the caller.  I think emitting an event here would be nice.

Reply:
Event now in place